### PR TITLE
fix: vchart-all registration delayed execution

### PR DESCRIPTION
**Pull request by Gavinchen92**
[[中文版模板 / Chinese template](https://github.com/VisActor/VChart/blob/main/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [x] New feature
- [x] Bug fix
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Refactoring
- [ ] Update dependency
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Release
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Workflow
- [ ] Other (about what?)

### 💡 Background and solution

Tree-shaking 通常只在生产环境启用以提升性能，这会导致按需加载组件在开发环境中未注册的功能因加载了 vchart-all 文件而能正常显示，隐藏了潜在错误，使得开发与生产环境行为不一致。改用懒加载注册器，只有在执行 new VChartAll 时才注册功能，能解决不一致问题。

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |           |
| 🇨🇳 Chinese |           |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [ ] Doc is updated/provided or not needed
- [ ] Demo is updated/provided or not needed
- [ ] TypeScript definition is updated/provided or not needed
- [ ] Changelog is provided or not needed

---

### 🚀 Summary

copilot:summary

### 🔍 Walkthrough

copilot:walkthrough

### CODE REVIEW

File: packages/vchart/src/vchart-all.ts [hunk 1]
```diff
@@ -69,7 +69,50 @@ import {
...
+    return VChart.registerMap(...args);
+  };
+
+  constructor(...args: ConstructorParameters<typeof VChart>) {
+    registerCache.forEach((fn: () => void) => {
+      if (typeof fn === 'function') {
+        // 确保元素是函数类型
+        fn();
+      } else {
+        console.error('Invalid function:', fn);
```
Comment (xile611) [inline packages/vchart/src/vchart-all.ts:104]: 不能直接 `console.error()`

File: packages/vchart/src/vchart-all.ts [hunk 2]
```diff
@@ -69,7 +69,50 @@ import {
...
 } from './interaction';
 
-VChart.useRegisters([
+const registerCache: (() => void)[] = [];
+
+/**
+ * VChartAll用于注册器的隔离
+ * @description Tree-shaking 通常只在生产环境启用以提升性能，导致按需加载组件在开发环境未注册的组件因加载了 vchart-all 文件而能正常显示，隐藏了潜在错误，使得开发与生产环境行为不一致。改用懒加载注册器，只有在执行 new VChartAll 时才注册组件，能解决不一致问题。
+ */
+class VChartAll extends VChart {
```
Comment (xile611) [inline packages/vchart/src/vchart-all.ts:78]: 封装看起来有点hack

File: packages/vchart/src/vchart-all.ts (hunk 1 above)
Comment (Gavinchen92) [reply packages/vchart/src/vchart-all.ts:104]: 我这是直接copy的VChart的代码，这里可以改成直接调用VChart.useRegisters(....)

File: packages/vchart/src/vchart-all.ts (hunk 2 above)
Comment (Gavinchen92) [reply packages/vchart/src/vchart-all.ts:78]: 确实有点hack, VChartAll和VChart的部分逻辑不一致，可能以后会有坑。但是我想不到更好的方法了

**Comment by Gavinchen92** [comment]
实现方法比较hack，可能会埋坑。close了

**Pull request closed without merging**